### PR TITLE
[Tools] add new tool to support Azure OpenAI GPT-4 Turbo with Vision

### PR DESCRIPTION
**Pull request by melionel**
- Description

This pull request introduces a new tool called "Azure OpenAI GPT-4 Turbo with Vision" to the codebase. It includes changes to multiple files to add support for analyzing images and answering questions about them using Azure OpenAI GPT-4 Turbo with Vision. The most important changes include adding the new tool to the `aoai_gpt4v.yaml` file, adding a test case to check image input support, and adding documentation for the new tool.

- Validation result:

*local*:
image
image

*cloud*
image
image
image

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-tools/promptflow/tools/aoai_gpt4v.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+try:
+    from openai import AzureOpenAI as AzureOpenAIClient
+except Exception:
+    raise Exception(
+        "Please upgrade your OpenAI package to version 1.0.0 or later using the command: pip install --upgrade openai.")
+
+from promptflow._internal import ToolProvider, tool
```
Comment (chjinche) [inline src/promptflow-tools/promptflow/tools/aoai_gpt4v.py:7]: need some uts for this tool?

**Review by chjinche — APPROVED** [review]

File: src/promptflow-tools/promptflow/tools/aoai_gpt4v.py (hunk 1 above)
Comment (melionel) [reply src/promptflow-tools/promptflow/tools/aoai_gpt4v.py:7]: add two test cases and skip for now

File: src/promptflow-tools/tests/test_aoai_gptv.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from promptflow.tools.aoai_gpt4v import AzureOpenAI
+
+
+@pytest.fixture
+def azure_openai_provider(azure_open_ai_connection) -> AzureOpenAI:
+    return AzureOpenAI(azure_open_ai_connection)
+
+
+@pytest.mark.usefixtures("use_secrets_config_file")
+@pytest.mark.skip_if_no_api_key("azure_open_ai_connection")
```
Comment (chjinche) [inline src/promptflow-tools/tests/test_aoai_gptv.py:12]: looks no need?

Comment (melionel) [reply src/promptflow-tools/tests/test_aoai_gptv.py]: yes, should remove

**Review by chjinche — APPROVED** [review]

**Review by DaweiCai — APPROVED** [review]

**Pull request merged**